### DISCUSSION TITLE
how to install astra-toolbox in vs-code?

### DISCUSSION THREAD

**Discussion by gmlwns001214** [community/community#83353] [Programming Help]
### Body

i wanna install astra-toolbox in vs code,but following messege error occured.. I took a look many solution.but i couldn't.. 
pip upgrade was installed. Should I use conda method?

python: 3.6.0

C:\Users\강희준>pip install astra-toolbox
Collecting astra-toolbox
  Using cached astra-toolbox-1.8b5.tar.gz (452 kB)
  Preparing metadata (setup.py) ... done
Requirement already satisfied: numpy in c:\users\강희준\appdata\local\programs\python\python36\lib\site-packages (from astra-toolbox) (1.19.5)
Requirement already satisfied: six in c:\users\강희준\appdata\local\programs\python\python36\lib\site-packages (from astra-toolbox) (1.16.0)  
Requirement already satisfied: scipy in c:\users\강희준\appdata\local\programs\python\python36\lib\site-packages (from astra-toolbox) (1.5.4) 
Requirement already satisfied: cython in c:\users\강희준\appdata\local\programs\python\python36\lib\site-packages (from astra-toolbox) (3.0.7)
Building wheels for collected packages: astra-toolbox
  Building wheel for astra-toolbox (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: 'c:\users\강희준\appdata\local\programs\python\python36\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Public\\Documents\\ESTsoft\\CreatorTemp\\pip-install-5kgfuntb\\astra-toolbox_ff3cefbdbaa24300add1745a7128ab7f\\setup.py'"'"'; __file__='"'"'C:\\Users\\Public\\Documents\\ESTsoft\\CreatorTemp\\pip-install-5kgfuntb\\astra-toolbox_ff3cefbdbaa24300add1745a7128ab7f\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d 'C:\Users\Public\Documents\ESTsoft\CreatorTemp\pip-wheel-bhr8ot2s'
       cwd: C:\Users\Public\Documents\ESTsoft\CreatorTemp\pip-install-5kgfuntb\astra-toolbox_ff3cefbdbaa24300add1745a7128ab7f\
  Complete output (11 lines):
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "C:\Users\Public\Documents\ESTsoft\CreatorTemp\pip-install-5kgfuntb\astra-toolbox_ff3cefbdbaa24300add1745a7128ab7f\setup.py", line 76, in <module>
      subprocess.call(['./autogen.sh'])
    File "c:\users\강희준\appdata\local\programs\python\python36\lib\subprocess.py", line 267, in call
      with Popen(*popenargs, **kwargs) as p:
    File "c:\users\강희준\appdata\local\programs\python\python36\lib\subprocess.py", line 707, in __init__
      restore_signals, start_new_session)
    File "c:\users\강희준\appdata\local\programs\python\python36\lib\subprocess.py", line 990, in _execute_child
      startupinfo)
  OSError: [WinError 193] %1은(는) 올바른 Win32 응용 프로그램이 아닙니다
  ----------------------------------------
  ERROR: Failed building wheel for astra-toolbox
  Running setup.py clean for astra-toolbox
  ERROR: Command errored out with exit status 1:
   command: 'c:\users\강희준\appdata\local\programs\python\python36\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Public\\Documents\\ESTsoft\\CreatorTemp\\pip-install-5kgfuntb\\astra-toolbox_ff3cefbdbaa24300add1745a7128ab7f\\setup.py'"'"'; __file__='"'"'C:\\Users\\Public\\Documents\\ESTsoft\\CreatorTemp\\pip-install-5kgfuntb\\astra-toolbox_ff3cefbdbaa24300add1745a7128ab7f\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' clean --all
       cwd: C:\Users\Public\Documents\ESTsoft\CreatorTemp\pip-install-5kgfuntb\astra-toolbox_ff3cefbdbaa24300add1745a7128ab7f
  Complete output (11 lines):
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "C:\Users\Public\Documents\ESTsoft\CreatorTemp\pip-install-5kgfuntb\astra-toolbox_ff3cefbdbaa24300add1745a7128ab7f\setup.py", line 76, in <module>
      subprocess.call(['./autogen.sh'])
    File "c:\users\강희준\appdata\local\programs\python\python36\lib\subprocess.py", line 267, in call
      with Popen(*popenargs, **kwargs) as p:
    File "c:\users\강희준\appdata\local\programs\python\python36\lib\subprocess.py", line 707, in __init__
      restore_signals, start_new_session)
    File "c:\users\강희준\appdata\local\programs\python\python36\lib\subprocess.py", line 990, in _execute_child
      startupinfo)
  OSError: [WinError 193] %1은(는) 올바른 Win32 응용 프로그램이 아닙니다
  ----------------------------------------
  ERROR: Failed cleaning build dir for astra-toolbox
Failed to build astra-toolbox
Installing collected packages: astra-toolbox
    Running setup.py install for astra-toolbox ... error
    ERROR: Command errored out with exit status 1:
     command: 'c:\users\강희준\appdata\local\programs\python\python36\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Public\\Documents\\ESTsoft\\CreatorTemp\\pip-install-5kgfuntb\\astra-toolbox_ff3cefbdbaa24300add1745a7128ab7f\\setup.py'"'"'; __file__='"'"'C:\\Users\\Public\\Documents\\ESTsoft\\CreatorTemp\\pip-install-5kgfuntb\\astra-toolbox_ff3cefbdbaa24300add1745a7128ab7f\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\Public\Documents\ESTsoft\CreatorTemp\pip-record-tksgojh4\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\users\강희준\appdata\local\programs\python\python36\Include\astra-toolbox'
         cwd: C:\Users\Public\Documents\ESTsoft\CreatorTemp\pip-install-5kgfuntb\astra-toolbox_ff3cefbdbaa24300add1745a7128ab7f\
    Complete output (11 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Users\Public\Documents\ESTsoft\CreatorTemp\pip-install-5kgfuntb\astra-toolbox_ff3cefbdbaa24300add1745a7128ab7f\setup.py", line 76, in <module>
        subprocess.call(['./autogen.sh'])
      File "c:\users\강희준\appdata\local\programs\python\python36\lib\subprocess.py", line 267, in call
        with Popen(*popenargs, **kwargs) as p:
      File "c:\users\강희준\appdata\local\programs\python\python36\lib\subprocess.py", line 707, in __init__
        restore_signals, start_new_session)
      File "c:\users\강희준\appdata\local\programs\python\python36\lib\subprocess.py", line 990, in _execute_child
        startupinfo)
    OSError: [WinError 193] %1은(는) 올바른 Win32 응용 프로그램이 아닙니다
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'c:\users\강희준\appdata\local\programs\python\python36\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Public\\Documents\\ESTsoft\\CreatorTemp\\pip-install-5kgfuntb\\astra-toolbox_ff3cefbdbaa24300add1745a7128ab7f\\setup.py'"'"'; __file__='"'"'C:\\Users\\Public\\Documents\\ESTsoft\\CreatorTemp\\pip-install-5kgfuntb\\astra-toolbox_ff3cefbdbaa24300add1745a7128ab7f\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\Public\Documents\ESTsoft\CreatorTemp\pip-record-tksgojh4\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\users\강희준\appdata\local\programs\python\python36\Include\astra-toolbox' Check the logs for full command output.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by dmpapageo** [comment]
I did some research based on your error and came across some possible solutions:


- Python Version Compatibility: The version of Python being used is 3.6. It's possible that astra-toolbox requires a newer version of Python. Upgrading to a more recent version (like Python 3.8 or 3.9) might resolve the issue. Remember to reinstall all necessary libraries after upgrading Python.

- Check for 32-bit vs 64-bit Python: The error message %1은(는) 올바른 Win32 응용 프로그램이 아닙니다 suggests a compatibility issue, possibly between 32-bit and 64-bit versions. If the Python installation or the astra-toolbox package is of a different architecture (32-bit vs 64-bit), this kind of error can occur. Ensure that both Python and the toolbox are either 32-bit or 64-bit.

- Windows Subsystem for Linux (WSL): Given that the error occurs in a script invoking a Unix-style ./autogen.sh, it might be that astra-toolbox is better suited for a Unix-like environment. If you are on Windows 10 or higher, consider using Windows Subsystem for Linux (WSL) for installation.

- Dependencies and Environment: Ensure all dependencies are correctly installed. Sometimes, missing or outdated dependencies can cause installation failures. Additionally, using a virtual environment (like venv or conda) can help isolate dependencies and versions for your project.

- Manual Installation: If the above steps don't work, consider manually installing astra-toolbox. This can be done by cloning the repository from GitHub (if available) and following the manual installation instructions usually provided in the README file.

I hope any of these help 🙏
